### PR TITLE
eslint rules part 1

### PR DESCRIPTION
**Pull request by sahilmob**

### CODE REVIEW

**Comment by steve8708** [comment]
This is awesome @sahilmob! @samijaber will take a look shortly. And in the meantime looks like you just need to run prettier to pass the build

From the project root:
```
npm install # if you haven't
npm run prettier -- --write .
```

then push up the changes

**Review by samijaber — CHANGES_REQUESTED** [review]
@sahilmob we had some issues with our CI: a `test` script was missing. I just merged https://github.com/BuilderIO/mitosis/pull/191. If you can pull the changes, your new tests will properly run in Github Actions now.

I manually ran `npx jest` locally and saw that one of the tests is failing (`use-state-var-declarator`). I'll finish looking over the other rules while you find what's wrong with that one :)

File: packages/eslint-plugin/src/rules/__tests__/no-var-declaration-in-jsx.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            <div>
+              {a.map(x =>{
+                return <span>{x}</span>
+              })}
+            </div>
+        );
+      }
+    `,
+    },
+    // Doesn't affect variable declaration inside JXS attributes.
```
Comment (samijaber) [inline packages/eslint-plugin/src/rules/__tests__/no-var-declaration-in-jsx.ts:34]:
```suggestion
    // Doesn't affect variable declaration inside JSX attributes.
```

File: packages/eslint-plugin/src/rules/__tests__/css-no-vars.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      jsx: true, // Allows for the parsing of JSX
+    },
+  },
+} as const;
+
+var ruleTester = new RuleTester();
+
+ruleTester.run('css-no-vars', rule, {
+  valid: [
+    { ...opts, code: '<button/>', ...opts },
```
Comment (samijaber) [inline packages/eslint-plugin/src/rules/__tests__/css-no-vars.ts:19]:
```suggestion
    { ...opts, code: '<button/>' },
```

File: packages/eslint-plugin/src/rules/only-default-function-and-imports.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+          if (
+            !types.isImportDeclaration(child) &&
+            !types.isExportDefaultDeclaration(child) &&
+            !types.isTypeAlias(child) &&
+            !types.isInterfaceDeclaration(child)
+          ) {
+            context.report({
+              node: child as any,
+              message:
+                "Mitosis component file shouldn't contain anything other than import declamations, the component itself a default export or type declarations",
```
Comment (samijaber) [inline packages/eslint-plugin/src/rules/only-default-function-and-imports.ts:48]:
```suggestion
                "Mitosis component files should only contain import declarations, the component itself (in a default export), and type declarations",
```T

File: packages/eslint-plugin/src/rules/use-state-var-declarator.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                id: {
+                  name: not('state'),
+                },
+              },
+            },
+            (node) => {
+              context.report({
+                node: node.parent.id as any,
+                message:
+                  'useState should be assigned to a variable called state exclusively',
```
Comment (samijaber) [inline packages/eslint-plugin/src/rules/use-state-var-declarator.ts:54]:
```suggestion
                  'useState should be exclusively assigned to a variable called state',
```

**Review by samijaber — COMMENTED** [review]
Looks fairly good. A few things to tweak, should be good afterwards :)

**Review by samijaber — COMMENTED** [review]
@sahilmob I forgot to mention this in the initial review, but we also need some documentation on what these rules represent. See https://github.com/BuilderIO/mitosis/blob/main/packages/eslint-plugin/docs/rules/jsx-callback-arg-name.md as an example for the one we already have.

Should be able to quickly put those together based on the tests & descriptions you've written 🙏🏽

**Comment by steve8708** [comment]
this is awesome @sahilmob, looks great!

**Comment by sahilmob** [comment]
Thanks. Looking forward to more contributions. 🚀 🚀 🚀

**Pull request merged**